app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,39 @@
+class ReviewsController < ApplicationController
+  before_action :set_user
+
+  def index
+    if params[:query].present?
+      author = User.where("name ILIKE ?", "%#{params[:query]}%").first
+      @reviews = Review.where(receiver: @user, author: author)
+    else
+      @reviews = Review.where(receiver: @user)
+    end
+  end
+
+  def new
+    @review = Review.new
+  end
+
+  def create
+    @review = Review.new(review_params)
+    review.author = current_user
```

Comment:
Before saving, we need to also set the review.receiver and review.barber_review, which have been defined in the #new action. The form (on the new view) that triggers the request to #create should have hidden fields containing the :barber_review and the :receiver.
So before saving you wanna set:
@review.barber_review = params[:barber_review]
@review.receiver = params[:receiver]